### ISSUE TITLE
Suggestion: Ability to set automatic smart-tax.

### ISSUE THREAD

**Issue by ElsaTheHobo** [TownyAdvanced/Towny#2835]
Thanks for all the work you guys do on towny, its my favorite plugin and I love to see it still in development after so many years!
Here are my suggestions:

~~1: Allow taxpercent for nations~~ Covered in #3181 
2: A new tax mode: **The mayor/king specifies the amount they want as a goal for the day and that tax is divided between all members.**

- A mayor wants 100$ a day with 7 residents? Each resident has to pay $14.29ish
- This could also be done tax percent wise, idk which one would be better (or maybe even make them two new separate tax modes??) So instead, if a resident has 50% of the sum of all the residents's money, then that resident would have to pay 50% of the tax.
- The value could also be set to 'auto' to match upkeep (plus nationtax if its a town in a nation)
- AND, as an extra special bonus, the tax syntax could be **auto+number**

Example: A town has a daily upkeep of 20 and a nationtax of 5. The value for 'auto' is then 25. But the mayor also wants the town to gain a lil $$ for claiming land and whatever else towns use money for. The mayor can do **/t set tax auto+5 taxgoal**. Now, the town would earn 5$ each day after taxes (30 before tax). And, if the nationtax or daily upkeep changes, the auto value would update too

This would probably mean getting rid of **/tn toggle taxpercent**. Instead, a new command could be created, with the syntax **/tn set tax _[value or expression]_ _[taxtype]_**
The taxtypes are just {**default, taxpercent, taxgoal, taxgoalpercent**} and is optional (does not change to default if left out, just stays as is)

Tl;dr new tax mode that allows mayors/kings to be lazy and just say how much they want total
